chore: add examples for aria labels for Input (#1365)
ember-learn/guides-source: guides/release/components/built-in-components.md

<label for="site">Ember Question</label>
<input id="site" type="text" value="How do text fields work?"/>
```

> ### Naming with aria-label

Reviewer: I think maybe we want to remove this section and only show "naming with aria-labelledby. What do you think @MelSumner?

Reviewer 2: fwiw, that is my understanding as well and I agree.

Reviewer 3: I think it would be good to have "ways to associate labels" and then use/explain all of them.

Reviewer 4: Little ping here for us to revisit this after we put up the Accessibility Spotlight blog posts perhaps. :)

The aria-label property enables developers to name an element with a string that is not visually rendered.

```handlebars
<Input id="site" @value="How do text fields work?" aria-label="Ember Question"/>
```

The aria-label property is useful when there is no visible text content that will serve as an appropriate accessible name. 

### Naming with aria-labelledby
The aria-labelledby property enables developers to reference other elements on the page to define an accessible name. 

```handlebars
<span id="site-label">Ember Question</span>

> <Input id="site" @value="How do text fields work?" aria-labelledby="site-label"/>

Reviewer: @MelSumner Would it make sense for this to have both a regular label and a labelledby or describedby?

Reviewer 3: `aria-labelledby` is best used for search inputs. I'll suggest some changes to the example.

```

You can pass the following standard `<input>` attributes within the input

> helper:

Reviewer: ```suggestion
Here are some other standard `<input>` attributes that are supported:
```

<label for="site">Ember Question</label>
<input id="site" type="text" value="How do text fields work?"/>
```

> ### Naming with aria-label

Reviewer: @MelSumner how about something like this?

> The `aria-label` property enables developers to name an element with a string that is not visually rendered.

Reviewer: @rwjblue has some feedback on when to use property vs attribute in the prose. I mix them up all the time and I'm not sure which is correct in these examples.

```handlebars
<Input id="site" @value="How do text fields work?" aria-label="Ember Question"/>
```

The `aria-label` property is useful when there is no visible text content that will serve as an appropriate accessible name. 

### Setting attributes on Input

Just like a native `<input>` element, there are many different types of attributes that you can apply to Ember's `<Input />` component, such as `aria` attributes or `required`. 

> For example, the `aria-labelledby` property enables developers to reference other elements on the page to define an accessible name:

Reviewer 3: ```suggestion
For example, the `aria-labelledby` property is useful for situations like a search input, where the search button can serve as the label for the input element:
```

```handlebars

> <span id="site-label">Ember Question</span>
> <Input id="site" @value="How do text fields work?" aria-labelledby="site-label"/>

Reviewer 3: ```suggestion
<Input aria-labelledby="button-search" />
<button id="button-search">Search</button>
```

```

> Here are some other standard `<input>` attributes that are supported:

Reviewer 5: Some of the lists just below are attributes and others are arguments (e.g. `@value`).

```suggestion
Here are some other standard `<input>` attributes and arguments that are supported:
```

<label for="site">Ember Question</label>
<input id="site" type="text" value="How do text fields work?"/>
```
### Ways to associate labels and inputs

Every input should be associated with a label. Within HTML, there are several different ways to do this.  In this section, we will show how to apply those strategies for Ember inputs.

You can nest the input inside the label:

<label>
    Ask a question about Ember:
    <Input type="text" @value={{this.val}} />
</label>

You can associate the label using `for` and `id`:

<label for={{this.myUniqueId}}>
    Ask a question about Ember:
</label>
<Input id={{this.myUniqueId}} type="text" @value={{this.val}} />

> The `aria-label` property enables developers to name an element with a string that is not visually rendered.

Reviewer 3: ```suggestion
The `aria-label` attribute enables developers to label an input element with a string that is not visually rendered, but still available to assistive technology. 
```

```handlebars
<Input id="site" @value="How do text fields work?" aria-label="Ember Question"/>
```

> The `aria-label` property is useful when there is no visible text content that will serve as an appropriate accessible name. 

Reviewer 3: ```suggestion
While it is more appropriate to use a `<label>` element, the `aria-label` attribute can be used in instances where visible text content is not possible.
```

### Setting attributes on Input

> Just like a native `<input>` element, there are many different types of attributes that you can apply to Ember's `<Input />` component, such as `aria` attributes or `required`. 

Reviewer 3: ```suggestion
Just like a native `<input>` element, there are many different types of attributes that you can apply to Ember's `<Input />` component, such as the `aria-*` attributes or `required`. 
```

```handlebars
<label>
    Ask a question about Ember:
    <Input type="text" @value={{this.val}} />
</label>
```

You can associate the label using `for` and `id`:

```handlebars
<label for={{this.myUniqueId}}>
    Ask a question about Ember:
</label>
<Input id={{this.myUniqueId}} type="text" @value={{this.val}} />
```

The `aria-label` attribute enables developers to label an input element with a string that is not visually rendered, but still available to assistive technology. 

```handlebars
<Input id="site" @value="How do text fields work?" aria-label="Ember Question"/>
```

While it is more appropriate to use a `<label>` element, the `aria-label` attribute can be used in instances where visible text content is not possible.

### Setting attributes on Input

Just like a native `<input>` element, there are many different types of attributes that you can apply to Ember's `<Input />` component, such as the `aria-*` attributes or `required`. 
For example, the `aria-labelledby` property is useful for situations like a search input, where the search button can serve as the label for the input element:

```handlebars
<Input aria-labelledby="button-search" />

> <button id="button-search">Search</button>

Reviewer 6: Could we specify the button type, to complement Matthew's work in https://github.com/ember-learn/guides-source/pull/1425?

```suggestion
<button id="button-search" type="button">Search</button>
```
